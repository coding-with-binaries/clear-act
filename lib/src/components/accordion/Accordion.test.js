import {
  Accordion,
  AccordionContent,
  AccordionDescription,
  AccordionPanel,
  AccordionTitle
} from './Accordion';

import React from 'react';
import renderer from 'react-test-renderer';

test('Renders basic accordion', () => {
  const component = renderer.create(
    <Accordion>
      <AccordionPanel
        accordionTitle={
          <AccordionTitle>Basic Accordion Title 1</AccordionTitle>
        }
        accordionContent={
          <AccordionContent>Basic Accordion Content 1</AccordionContent>
        }
      />
      <AccordionPanel
        accordionTitle={
          <AccordionTitle>Basic Accordion Title 2</AccordionTitle>
        }
        accordionContent={
          <AccordionContent>Basic Accordion Content 2</AccordionContent>
        }
      />
    </Accordion>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders accordion with description', () => {
  const component = renderer.create(
    <Accordion>
      <AccordionPanel
        accordionTitle={<AccordionTitle>Accordion Title 1</AccordionTitle>}
        accordionContent={
          <AccordionContent>Accordion Content 1</AccordionContent>
        }
        accordionDescription={
          <AccordionDescription>Accordion Description 1</AccordionDescription>
        }
      />
      <AccordionPanel
        accordionTitle={<AccordionTitle>Accordion Title 2</AccordionTitle>}
        accordionContent={
          <AccordionContent>Accordion Content 2</AccordionContent>
        }
        accordionDescription={
          <AccordionDescription>Accordion Description 2</AccordionDescription>
        }
      />
    </Accordion>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders multi-panel accordion', () => {
  const component = renderer.create(
    <Accordion accordionMultiPanel>
      <AccordionPanel
        accordionTitle={
          <AccordionTitle>Multi Panel Accordion Title 1</AccordionTitle>
        }
        accordionContent={
          <AccordionContent>Multi Panel Accordion Content 1</AccordionContent>
        }
      />
      <AccordionPanel
        accordionTitle={
          <AccordionTitle>Multi Panel Accordion Title 2</AccordionTitle>
        }
        accordionContent={
          <AccordionContent>Multi Panel Accordion Content 2</AccordionContent>
        }
      />
    </Accordion>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
