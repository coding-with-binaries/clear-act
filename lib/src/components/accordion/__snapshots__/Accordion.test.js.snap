// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders accordion with description 1`] = `
<div
  className="clr-accordion"
>
  <div
    className="clr-accordion-panel"
  >
    <div
      className="clr-accordion-panel-inactive"
      role="group"
    >
      <div
        className="clr-accordion-header"
      >
        <button
          className="clr-accordion-header-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="clr-accordion-status"
          >
            <clr-icon
              class="clr-accordion-angle"
              dir="right"
              shape="angle"
            />
            <span
              className="clr-accordion-number"
            />
            <clr-icon
              class="clr-accordion-error-icon"
              shape="exclamation-circle"
            />
            <clr-icon
              class="clr-accordion-complete-icon"
              shape="check-circle"
            />
          </span>
          <div
            className="clr-accordion-title"
          >
            Accordion Title 1
          </div>
          <div
            className="clr-accordion-description"
          >
            Accordion Description 1
          </div>
        </button>
      </div>
      <div
        role="region"
      >
        <div
          className="clr-accordion-content"
          style={
            Object {
              "height": "auto",
            }
          }
        >
          <div
            className="clr-accordion-inner-content"
          >
            Accordion Content 1
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="clr-accordion-panel"
  >
    <div
      className="clr-accordion-panel-inactive"
      role="group"
    >
      <div
        className="clr-accordion-header"
      >
        <button
          className="clr-accordion-header-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="clr-accordion-status"
          >
            <clr-icon
              class="clr-accordion-angle"
              dir="right"
              shape="angle"
            />
            <span
              className="clr-accordion-number"
            />
            <clr-icon
              class="clr-accordion-error-icon"
              shape="exclamation-circle"
            />
            <clr-icon
              class="clr-accordion-complete-icon"
              shape="check-circle"
            />
          </span>
          <div
            className="clr-accordion-title"
          >
            Accordion Title 2
          </div>
          <div
            className="clr-accordion-description"
          >
            Accordion Description 2
          </div>
        </button>
      </div>
      <div
        role="region"
      >
        <div
          className="clr-accordion-content"
          style={
            Object {
              "height": "auto",
            }
          }
        >
          <div
            className="clr-accordion-inner-content"
          >
            Accordion Content 2
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders basic accordion 1`] = `
<div
  className="clr-accordion"
>
  <div
    className="clr-accordion-panel"
  >
    <div
      className="clr-accordion-panel-inactive"
      role="group"
    >
      <div
        className="clr-accordion-header"
      >
        <button
          className="clr-accordion-header-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="clr-accordion-status"
          >
            <clr-icon
              class="clr-accordion-angle"
              dir="right"
              shape="angle"
            />
            <span
              className="clr-accordion-number"
            />
            <clr-icon
              class="clr-accordion-error-icon"
              shape="exclamation-circle"
            />
            <clr-icon
              class="clr-accordion-complete-icon"
              shape="check-circle"
            />
          </span>
          <div
            className="clr-accordion-title"
          >
            Basic Accordion Title 1
          </div>
        </button>
      </div>
      <div
        role="region"
      >
        <div
          className="clr-accordion-content"
          style={
            Object {
              "height": "auto",
            }
          }
        >
          <div
            className="clr-accordion-inner-content"
          >
            Basic Accordion Content 1
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="clr-accordion-panel"
  >
    <div
      className="clr-accordion-panel-inactive"
      role="group"
    >
      <div
        className="clr-accordion-header"
      >
        <button
          className="clr-accordion-header-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="clr-accordion-status"
          >
            <clr-icon
              class="clr-accordion-angle"
              dir="right"
              shape="angle"
            />
            <span
              className="clr-accordion-number"
            />
            <clr-icon
              class="clr-accordion-error-icon"
              shape="exclamation-circle"
            />
            <clr-icon
              class="clr-accordion-complete-icon"
              shape="check-circle"
            />
          </span>
          <div
            className="clr-accordion-title"
          >
            Basic Accordion Title 2
          </div>
        </button>
      </div>
      <div
        role="region"
      >
        <div
          className="clr-accordion-content"
          style={
            Object {
              "height": "auto",
            }
          }
        >
          <div
            className="clr-accordion-inner-content"
          >
            Basic Accordion Content 2
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders multi-panel accordion 1`] = `
<div
  className="clr-accordion"
>
  <div
    className="clr-accordion-panel"
  >
    <div
      className="clr-accordion-panel-inactive"
      role="group"
    >
      <div
        className="clr-accordion-header"
      >
        <button
          className="clr-accordion-header-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="clr-accordion-status"
          >
            <clr-icon
              class="clr-accordion-angle"
              dir="right"
              shape="angle"
            />
            <span
              className="clr-accordion-number"
            />
            <clr-icon
              class="clr-accordion-error-icon"
              shape="exclamation-circle"
            />
            <clr-icon
              class="clr-accordion-complete-icon"
              shape="check-circle"
            />
          </span>
          <div
            className="clr-accordion-title"
          >
            Multi Panel Accordion Title 1
          </div>
        </button>
      </div>
      <div
        role="region"
      >
        <div
          className="clr-accordion-content"
          style={
            Object {
              "height": "auto",
            }
          }
        >
          <div
            className="clr-accordion-inner-content"
          >
            Multi Panel Accordion Content 1
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="clr-accordion-panel"
  >
    <div
      className="clr-accordion-panel-inactive"
      role="group"
    >
      <div
        className="clr-accordion-header"
      >
        <button
          className="clr-accordion-header-button"
          onClick={[Function]}
          type="button"
        >
          <span
            className="clr-accordion-status"
          >
            <clr-icon
              class="clr-accordion-angle"
              dir="right"
              shape="angle"
            />
            <span
              className="clr-accordion-number"
            />
            <clr-icon
              class="clr-accordion-error-icon"
              shape="exclamation-circle"
            />
            <clr-icon
              class="clr-accordion-complete-icon"
              shape="check-circle"
            />
          </span>
          <div
            className="clr-accordion-title"
          >
            Multi Panel Accordion Title 2
          </div>
        </button>
      </div>
      <div
        role="region"
      >
        <div
          className="clr-accordion-content"
          style={
            Object {
              "height": "auto",
            }
          }
        >
          <div
            className="clr-accordion-inner-content"
          >
            Multi Panel Accordion Content 2
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
